inc/integrations/class-auth.php

```diff
@@ -0,0 +1,258 @@
+	 *
+	 * @param string $password Application password.
+	 * @return string
+	 */
+	public function get_formatted_token_cookie( $password ) : string {
+		$user = wp_get_current_user();
+
+		if ( is_wp_error( $user ) ) {
+			return '';
+		}
+
+		return base64_encode( $user->data->user_login . ':' . str_replace( ' ', '', $password ) );
+	}
+
+	/**
+	 * If the user isn't logged in. but has an auth cookie, kill it.
+	 *
+	 * @todo what other checks do we need in here that would trigger removing the cookie? Last used?
+	 * @return bool Was the cookie was removed successfully?
+	 */
+	public function possibly_remove_cookie(): bool {
+		// Only care if we have a cookie.
+		if ( ! isset( $_COOKIE[ self::TOKEN_COOKIE_NAME ] ) ) { // phpcs:ignore
+			return false;
+		}
+
+		// Get active application passwords.
+		$passwords = \get_user_meta(
+			get_current_user_id(),
+			\WP_Application_Passwords::USERMETA_KEY_APPLICATION_PASSWORDS,
+			true
+		);
+
+		// If no passwords are stored for current user, remove cookies.
+		if ( empty( $passwords ) ) {
+			$this->remove_cookie();
+		}
+
+		$matching_passwords = array_filter(
+			$passwords,
+			function ( $password ) {
+				return (
+					! empty( $_COOKIE[ self::APP_ID_COOKIE_NAME ] ) && // phpcs:ignore
+					$password['app_id'] == $_COOKIE[ self::APP_ID_COOKIE_NAME ] // phpcs:ignore
+				);
+			}
+		);
+
+		// If no stored passwords match the curren App ID, remove cookies.
+		if ( empty( $matching_passwords ) ) {
+			$this->remove_cookie();
+		}
+
+		return false;
+	}
+
+	/**
+	 * Set the expiration on the cookie to unset it.
+	 */
+	public function remove_cookie() {
+		// phpcs:ignore
+		setcookie(
+			self::TOKEN_COOKIE_NAME,
+			null,
+			-1,
+			'/',
+			$this->cookie_domain
+		);
+
+		// phpcs:ignore
+		setcookie(
+			self::APP_ID_COOKIE_NAME,
+			null,
+			-1,
+			'/',
+			$this->cookie_domain
+		);
+	}
+
+	/**
+	 * Delete all old application passwords for the current user.
+	 *
+	 * @return void
+	 */
+	public function delete_all_application_passwords() {
+		// Set the new request with the new key and secret.
+		$app_pass_delete_request = new \WP_REST_Request(
+			\WP_REST_Server::DELETABLE,
+			'/wp/v2/users/me/application-passwords'
+		);
+
+		// Let's get the token.
+		rest_do_request( $app_pass_delete_request );
+	}
+
+	/**
+	 * Get or create an application password.
+	 *
+	 * @return array
+	 */
+	public function create_application_password() : array {
+		$user     = wp_get_current_user();
+		$app_name = get_bloginfo( 'name' ) . ' Irving App, user ' . $user->ID;
+
+		// Set the new request with the new key and secret.
+		$app_pass_request = new \WP_REST_Request(
+			\WP_REST_Server::CREATABLE,
+			'/wp/v2/users/me/application-passwords'
+		);
+		$app_pass_request->set_query_params(
+			[
+				'name'   => $app_name,
+				'app_id' => wp_generate_uuid4(),
+			]
+		);
+
+		// Let's get the token.
+		$result = rest_do_request( $app_pass_request );
+		$status = $result->status ?? 0;
```

Comment:
This is also available.

```suggestion
		$status = $result->get_status() ?? 0;
```